slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbSlashingProtection.java

```diff
@@ -126,6 +126,26 @@ private boolean checkAndInsertAttestation(
       }
     }
 
+    final Optional<UInt64> minimumSourceEpoch =
+        signedAttestationsDao.minimumSourceEpoch(h, validatorId);
+    if (minimumSourceEpoch.filter(minEpoch -> sourceEpoch.compareTo(minEpoch) < 0).isPresent()) {
+      LOG.warn(
+          "Attestation source epoch {} is below minimum existing attestation source epoch {}",
+          sourceEpoch,
+          minimumSourceEpoch.get());
+      return false;
+    }
+
+    final Optional<UInt64> minimumTargetEpoch =
+        signedAttestationsDao.minimumTargetEpoch(h, validatorId);
+    if (minimumTargetEpoch.filter(minEpoch -> targetEpoch.compareTo(minEpoch) <= 0).isPresent()) {
```

Comment:
reconfirming ... `targetEpoch <= minEpoch`

Reply:
Yes, for sourceEpoch it's < minEpoch but for targetEpoch it's <= minEpoch. Not what I expected either but that's what in the spec listed here https://hackmd.io/@sproul/Bk0Y0qdGD#Semantics

Reply:
In my head I saw:
if(minimumTargetEpoch.map(minEpoch -> targetEpoch.compareTo(minEpoch)).orElse(false)) {
...
}
But it is literally semantics

Reply:
and tbh - a function something like "epochLessThanThreshold(threshold, value)" would make this a BIT easier to read ... mainly because I hate the compareTo function.

Reply:
changed to use map